### PR TITLE
add provider checklist

### PR DESCRIPTION
**Pull request by stevendborrelli**
Signed-off-by: Steven Borrelli <steve@borrelli.org>

### Description of your changes

This PR adds a provider development checklist, documenting common and recommended practices for structuring projects.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

I ran `make reviewable test`. No code changes are present in this PR.

### CODE REVIEW

File: PROVIDER_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Provider Checklist
```
Comment (negz) [inline PROVIDER_CHECKLIST.md:1]: Nit: Would you mind wrapping this at 80 cols? It's the convention for docs in this repo. https://marketplace.visualstudio.com/items?itemName=stkb.rewrap can automate it.

File: PROVIDER_CHECKLIST.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+It is recommended to use the [crossplane-contrib](https://github.com/orgs/crossplane-contrib) Github Organization. Your project should be named `provider-<name>`. Members of the Crossplane community are happy to create a repository and configure access. Please reach out to the Crossplane [#dev](https://crossplane.slack.com/archives/CEF5N8X08) channel.
+
+There is a template repository called [provider-template](https://github.com/crossplane/provider-template) that can be used as a starting point for new providers.
+
+## Files
+
+Most Crossplane providers include the following files:
+
+- [ ]  A descriptive README.md at the root of the project (see [provider-gcp/README.md](https://github.com/crossplane/provider-gcp/blob/master/README.md) as an example)
+- [ ]  Code is licensed under the[ Apache 2.0 License](https://github.com/crossplane/provider-template/blob/main/LICENSE)
```
Comment (negz) [inline PROVIDER_CHECKLIST.md:20]:
```suggestion
- [ ]  Code is licensed under the [Apache 2.0 License](https://github.com/crossplane/provider-template/blob/main/LICENSE)
```

File: PROVIDER_CHECKLIST.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+To build a new provider, please refer to the  [Provider Development Guide](https://crossplane.io/docs/master/contributing/provider_development_guide.html).
+
+This document contains list of items that are usually involved in managing Open Source contributions to a Crossplane provider. In general your provider should follow the guidelines documented in the [Crossplane Contributing](https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md) guide.
+
+## Repository
+
+It is recommended to use the [crossplane-contrib](https://github.com/orgs/crossplane-contrib) Github Organization. Your project should be named `provider-<name>`. Members of the Crossplane community are happy to create a repository and configure access. Please reach out to the Crossplane [#dev](https://crossplane.slack.com/archives/CEF5N8X08) channel.
```
Comment (negz) [inline PROVIDER_CHECKLIST.md:11]: I wonder if we should ask folks to raise an issue on this repo (as well as a Slack ping)? I find Slack pings often get lost when we're busy.

File: PROVIDER_CHECKLIST.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  - [ ] Install Provider
+  - [ ] Contribute to Development
+  - [ ] Authenticate to backend API, including creating Kubernetes secrets for the ProviderConfig
+- [ ] Include examples for the ProviderConfig and each resource in the `examples/` directory.
+
+## Build Process
+
+There are a number of build tools and processes that are common across the Crossplane ecosystem. Using these ensures a consistent development environment across projects.
+
+- [ ]  Use the [Upbound build](https://github.com/upbound/build) submodule. (see [https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#establishing-a-development-environment](https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#establishing-a-development-environment))
```
Comment (negz) [inline PROVIDER_CHECKLIST.md:35]: https://github.com/crossplane/provider-template/pull/19#issuecomment-987182490

I have an ancient PR that would make this the default. I'll try follow up this week.

**Review by negz — COMMENTED** [review]
Thanks for tackling this @stevendborrelli!

File: PROVIDER_CHECKLIST.md (hunk 3 above)
Comment (stevendborrelli) [reply PROVIDER_CHECKLIST.md:11]: What if we made this part of the workflow in the [crossplane/org ](https://github.com/crossplane/org)? We could have a new-repository issue template that includes many parts of this workflow?

File: PROVIDER_CHECKLIST.md (hunk 4 above)
Comment (stevendborrelli) [reply PROVIDER_CHECKLIST.md:35]: If you can't get to it, I will be happy to take a look. I've moved a couple of repos to the build submodule recently so it shouldn't take that long.

File: PROVIDER_CHECKLIST.md (hunk 3 above)
Comment (stevendborrelli) [reply PROVIDER_CHECKLIST.md:11]: Opened issue https://github.com/crossplane/org/issues/26

File: PROVIDER_CHECKLIST.md (hunk 4 above)
Comment (negz) [reply PROVIDER_CHECKLIST.md:35]: It's all updated now, and I think there's an appetite to merge it. Doing so would let us simplify this section to something like:

1. Use the build submodule.
2. If you're part of the crossplane-contrib org and want to push images to the crossplane org in Docker Hub please ask a steering committee member to grant your project access to the GitHub org scoped secrets.

File: PROVIDER_CHECKLIST.md (hunk 2 above)
Comment (stevendborrelli) [reply PROVIDER_CHECKLIST.md]: fixed

File: PROVIDER_CHECKLIST.md (hunk 4 above)
Comment (stevendborrelli) [reply PROVIDER_CHECKLIST.md]: I've updated the text and broke out the artifacts section.

**Comment by negz** [comment]
@stevendborrelli I'm going through some older issues and came across this one. If we end up merging https://github.com/crossplane/crossplane/pull/3045 (our updated charter) I could see this not being a great fit for the c/c repo, since we're going to be less opinionated about how providers are built and governed. I wonder if the provider-template repo would be a better place for this?

**Comment by elohmrow** [comment]
@negz i think the charter is needed, but is a different animal altogether. a big problem we're seeing across our customer base is there's simply not enough guidance.

today, our customers come to us after having completed some toy POC, looking for us to be SME, and we're not providing that value they are looking for. what i am seeing is folks are treating upbound like the professional services arm of crossplane.

this is one major point i was trying to make [here](https://docs.google.com/presentation/d/1aFsUKF-KCQiCC4CGKaVtxMMXmyA1ZPuQCKBF4pcY9EE/edit#slide=id.g11e29100eb0_0_16) and [here](https://upboundio.slack.com/archives/C031S3S6D33/p1650085903484049?thread_ts=1650009701.476089&cid=C031S3S6D33)

**Pull request closed without merging**